commit f3e8e6c6cf55139bbb8fd9031898c2f3add9d2c8 (HEAD -> master, refs/changes/95/3095/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-12-12T10:55:13-08:00 (1 year, 10 months ago)
    
    joyent/node-strsplit#4 strsplit loses the multiline and ignoreCase flags on a given RegExp pattern
    Reviewed by: David Pacheco <dap@joyent.com>
    Approved by: David Pacheco <dap@joyent.com>

diff --git a/lib/strsplit.js b/lib/strsplit.js
index 92dca0e..4b36ab2 100644
--- a/lib/strsplit.js
+++ b/lib/strsplit.js
@@ -6,7 +6,7 @@ module.exports.strsplit = strsplit;
 
 function strsplit(str, pattern, limit)
 {
-	var i, rv, last, match;
+	var i, rv, flags, last, match;
 
 	/*
 	 * If "limit" is not specified or is negative, our behavior is exactly
@@ -28,8 +28,14 @@ function strsplit(str, pattern, limit)
 		pattern = new RegExp(
 		    /* JSSTYLED */
 		    pattern.replace(/([.?*+^$[\]\\(){}|-])/g, "\\$1"), 'g');
-	else
-		pattern = new RegExp(pattern.source, 'g');
+	else {
+		flags = 'g';
+		if (pattern.multiline)
+			flags += 'm';
+		if (pattern.ignoreCase)
+			flags += 'i';
+		pattern = new RegExp(pattern.source, flags);
+	}
 
 	/*
 	 * The end condition is a bit unusual: if limit is 0 or negative, we're
diff --git a/tests/tst.strpatterns.js b/tests/tst.strpatterns.js
index f6a2a05..8b61c9a 100755
--- a/tests/tst.strpatterns.js
+++ b/tests/tst.strpatterns.js
@@ -35,3 +35,7 @@ mod_assert.deepEqual([ 'one', 'two three' ],
 /* no pattern is equivalent to \s+ */
 mod_assert.deepEqual([ 'one', 'two', 'three', 'four' ],
     strsplit('one \t two   three\t\nfour'));
+
+/* Don't lose flags when a limit is given. */
+mod_assert.deepEqual([ 'foo\n\n', 'bar' ],
+    strsplit('foo\n\n    bar', /^    /m, 2));
\ No newline at end of file
